charts/vector-aggregator/templates/haproxy/deployment.yaml

```diff
@@ -27,7 +27,9 @@ spec:
 {{ toYaml .Values.haproxy.podAnnotations | indent 8 }}
       {{- end }}
     spec:
+    {{- if .Values.haproxy.serviceAccount.create }}
       serviceAccountName: {{ include "libvector.serviceAccountName" . }}-haproxy
+    {{- end }}
```

Comment:
I think it's common to not wrap this in a conditional, see: https://github.com/haproxytech/helm-charts/blob/main/haproxy/templates/deployment.yaml#L52

Do you have any examples of this pattern in other charts?

Reply:
Hello, yes indeed this is not a very good solution, the problem is the name of the "ServiceAccountName" "-haproxy" in the deployment file.

When we don't want to activate the "ServiceAccount" in the haproxy, it will use the default "default" and add "-haproxy" at the end to give this name: (default-haproxy) and the replica will fail because it cannot find this serviceaccount.

The "default" value is supplied by the "libvector.serviceAccountName" function present in the "vector-shared" graph which has no condition on the creation value (true / false) of a "ServiceAccount"
Suddenly, it necessarily returns "default".

However, it would be interesting to use the haproxy.serviceAccountName function as for the "vector" graphic to correct this problem. :)

Reply:
Doh - oh yeah, much better solution 😄 I missed that completely.